kernel/async/monitor-common/async/Tasklet.hh

```diff
@@ -54,6 +59,18 @@ public:
 
   Tasklet(const Tasklet&) = delete;
 
+#ifdef NDEBUG
+    bool isInit() { return true; }
+    void setInit() {}
+    bool isUnused() { return true; }
+    void setUnused() {}
+#else
+    bool isInit() { return nextTasklet == INIT; }
+    void setInit() { nextTasklet = INIT; }
+    bool isUnused() { return nextTasklet == UNUSED; }
+    void setUnused() { nextTasklet = UNUSED; }
+#endif
+
   void run() {
     ASSERT(handler > FunPtr(VIRT_ADDR));
     handler(this);
```

Comment:
`      assert(isInit());       setUnused();`

and in set(): `assert(isUnused());      setInit();` and use const&& with the std::move: `new(payload) FUNCTOR(std::move(fun)); // copy-construct`

and in get(): `return MSG(std::move(caster->msg))`